src/NonBlocking/ConcurrentDictionary/DictionaryImplBoxed.cs

```diff
@@ -37,7 +37,7 @@ protected override bool TryClaimSlotForPut(ref Boxed<TKey> entryKey, TKey key)
                 }
             }
 
-            return _keyComparer.Equals(key, entryKey.Value);
+            return _keyComparer.Equals(key, entryKeyValue.Value);
```

Comment:
Unlike the case with primitives this does look like a bug. The use of `entryKey` is likely unintentional (see [the other comment](https://github.com/VSadov/NonBlocking/pull/23/files#r1256266434)), but for some time was not dangerous.  
The sweeping of removed keys came relatively late and introduced a case when an installed key may become `null`. 
I guess sweeping is very rare thus we have not seen this kind of race.

Good catch!